### PR TITLE
APP-14990 Refactor to make registering AJV schemas and formats more modular 

### PR DESCRIPTION
**Pull request by eyadmba**
Needed to run validations on entities in the persister, and that required different constructor parameters to be passed to the AJV instance, so I couldn't use the already existing one in `src/IntegrationSchema.ts`.

Instead, I exposed a function that takes in an AJV instance, and sets all the schemas on it. This way I can control the instance's constructor params, and just leverage the package's function to set the schemas on it.
Similar thing was done for the custom formats of AJV. There's a function that adds all the custom formats on an AJV instance.

So this is how I'd end up using those two new functions:
```typescript
const myInstance = new Ajv({ whatever I want });
registerSchemas(myInstance);
registerFormats(myInstance);
```

### CODE REVIEW

File: src/registerSchemas.ts [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+import UserJson from './schemas/User.json';
+import UserGroupJson from './schemas/UserGroup.json';
+import VaultJson from './schemas/Vault.json';
+import VendorJson from './schemas/Vendor.json';
+import VulnerabilityJson from './schemas/Vulnerability.json';
+import WeaknessJson from './schemas/Weakness.json';
+import WorkflowJson from './schemas/Workflow.json';
+import WorkloadJson from './schemas/Workload.json';
+
+export function registerSchemas(ajvInstance: Ajv): void {
```
Comment (eyadmba) [inline src/registerSchemas.ts:107]: As explained in the PR's description, this is what takes in an AJV instance and adds all the data model's schemas to it.

Comment (eyadmba) [reply src/registerSchemas.ts:107]: Similar to `src/IntegrationSchema.ts`, this file gets auto generated entirely using the script `tools/generate-register-schemas-function.sh`.

File: src/IntegrationSchema.ts [hunk 2]
```diff
@@ -24,14 +19,9 @@ const ipv6CidrRegex = /^([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}\/(12[0-8]|1[01]
...
-    ipCidr: (x) => ipv4CidrRegex.test(x) || ipv6CidrRegex.test(x),
-  },
 });
 
-// Install ajv-formats
-addFormats(IntegrationSchema);
+registerAjvFormats(IntegrationSchema);
 
 // Schema Imports : generated by tools/generate-schema-imports.sh
 import WorkloadJson from './schemas/Workload.json';
```
Comment (eyadmba) [inline src/IntegrationSchema.ts:27]: I could have replaced all this code with a simple call to the new function: `registerSchemas(IntegrationSchema)`, but there are exposed variables from this module that I didn't know if they're being used elsewhere or not, so I had to leave them.

File: src/registerAjvFormats.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  name: 'ip',
+  fn: (x) => ipv4.test(x) || ipv6.test(x),
+};
+
+const isValidIpCidrFormat = {
+  name: 'ipCidr',
+  fn: (x) => ipv4CidrRegex.test(x) || ipv6CidrRegex.test(x),
+};
+
+export function registerAjvFormats(ajvInstance: Ajv): void {
```
Comment (eyadmba) [inline src/registerAjvFormats.ts:19]: Similar to registerSchemas, here's `registerAjvFormats`.

File: src/registerFormats.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  name: 'ip',
+  fn: (x) => ipv4.test(x) || ipv6.test(x),
+};
+
+const isValidIpCidrFormat = {
+  name: 'ipCidr',
+  fn: (x) => ipv4CidrRegex.test(x) || ipv6CidrRegex.test(x),
+};
+
+export function registerFormats(ajvInstance: Ajv): void {
```
Comment (eyadmba) [inline src/registerFormats.ts:19]: Similar to `registerSchemas`, here's `registerFormats()`.

File: src/IntegrationSchema.ts [hunk 5]
```diff
@@ -24,14 +19,9 @@ const ipv6CidrRegex = /^([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}\/(12[0-8]|1[01]
...
-    ipCidr: (x) => ipv4CidrRegex.test(x) || ipv6CidrRegex.test(x),
-  },
 });
 
-// Install ajv-formats
-addFormats(IntegrationSchema);
+registerFormats(IntegrationSchema);
 
 // Schema Imports : generated by tools/generate-schema-imports.sh
 import WorkloadJson from './schemas/Workload.json';
```
Comment (eyadmba) [inline src/IntegrationSchema.ts:27]: I could have replaced all the code from this point until the end of the file with just a simple call to the new function: `registerSchemas(IntegrationSchema)`, but there are exposed members from this module in there that I didn't know if they're being used elsewhere or not, so I had to leave them.

**Review by zemberdotnet — DISMISSED** [review]

**Review by zemberdotnet — APPROVED** [review]

**Pull request merged**